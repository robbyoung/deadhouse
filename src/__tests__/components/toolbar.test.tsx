import React from 'react';
import renderer from 'react-test-renderer';
import Toolbar, {ToolbarItem} from '../../components/toolbar';

const testOptions: ToolbarItem[] = [
    {
        name: 'Tab 1',
        subItems: ['Option 1', 'Option 2'],
    },
    {
        name: 'Tab 2',
        subItems: ['Option 1'],
    },
    {
        name: 'Tab 3',
        subItems: [],
    },
];

describe('Toolbar component', () => {
    it('Can render a toolbar tab with no options', () => {
        const component = renderer.create(
            <Toolbar
                items={testOptions}
                startingIndex={2}
                onSelect={(): void => undefined}
            />,
        );

        expect(component).toMatchSnapshot();
    });

    it('Can render a toolbar tab with some options', () => {
        const component = renderer.create(
            <Toolbar
                items={testOptions}
                startingIndex={0}
                onSelect={(): void => undefined}
            />,
        );

        expect(component).toMatchSnapshot();
    });
});
