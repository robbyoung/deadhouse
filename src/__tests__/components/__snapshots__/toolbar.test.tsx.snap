// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component Can render a toolbar with basic navigation 1`] = `
Array [
  <div
    className="header"
  >
    <a
      className="home-link"
      href="/"
      onClick={[Function]}
    >
      <p
        className="title"
      >
        Kurald Galain
      </p>
      <p
        className="subtitle"
      >
        A Malazan Book of the Fallen Fansite
      </p>
    </a>
  </div>,
  ",",
  <div />,
]
`;
